codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/AsyncClientBuilderClass.java

```diff
@@ -91,7 +89,7 @@ public TypeSpec poetSpec() {
         }
 
         builder.addMethod(buildClientMethod());
-        builder.addMethod(initializeServiceClientConfigMethod());
+        builder.addMethod(SyncClientBuilderClass.initializeServiceClientConfigMethod(serviceConfigClassName));
```

Comment:
Using a static method of SyncClientBuilderClass within AsyncClientBuilderClass seems ambigous. Would it be better to move the initializeServiceClientConfigMethod to a common utility class used by both sync/Async ?

Reply:
Yes, we can do that in a follow up pull request if that's OK.

Reply:
Please add a TODO : here so that we might not miss it.

Reply:
I will remove this method and instead move it to the base class in the next PR.